app/models/patient.rb

```diff
@@ -28,6 +28,7 @@ class Patient
   before_save :save_identifier
 
   after_create :initialize_fulfillment
+  after_save :confirm_still_urgent
```

Comment:
I added this and i think it works, I'm guessing it is easier to control in the patient model than the module.
```
  before_save do
    self.urgent_flag = false unless still_urgent?
    true
  end
```